app/src/main/java/com/puzzlebench/clean_marvel_kotlin/presentation/mvp/CharacterDetailView.kt

```diff
@@ -0,0 +1,38 @@
+package com.puzzlebench.clean_marvel_kotlin.presentation.mvp
+
+import android.view.View
+import com.puzzlebench.clean_marvel_kotlin.R
+import com.puzzlebench.clean_marvel_kotlin.domain.model.Character
+import com.puzzlebench.clean_marvel_kotlin.presentation.extension.getImageByUrl
+import com.puzzlebench.clean_marvel_kotlin.presentation.extension.showToast
+import com.puzzlebench.clean_marvel_kotlin.presentation.fragment.DetailCharacterFragmentDialog
+import kotlinx.android.synthetic.main.fragment_dialog_detail_character.*
+import java.lang.ref.WeakReference
+
+class CharacterDetailView(fragment: DetailCharacterFragmentDialog) {
+    private val fragmentRef = WeakReference(fragment)
```

Comment:
Same here, please remove this weak reference and use the fragment you got as parameter